### DISCUSSION TITLE
Detailed configs?

### DISCUSSION THREAD

**Discussion by liguopeng0923** [yoshitomo-matsubara/torchdistill#411] [Q&A] [answered]
Hi,
Do you provide more details about your config file? I think it is not clear for any parameters.

Thank you.

**Comment by yoshitomo-matsubara** [comment]
Hi @liguopeng0923 

I need more contexts of your question. For instance, what config under `configs/` do you want to understand and what do you want to do using torchdistill?
Since torchdistill is built on PyTorch, you need to be familiar with PyTorch and its modules. In fact, many of the parameters and module names in config files can be found in PyTorch

**Comment by liguopeng0923** [comment]
Hi @yoshitomo-matsubara 


Thanks for your reply!!

For example, the mean of "key", "init".  
or where some parameters are used?
and the architecture of your config files. (what parameters should be used in specific situations?)

Overall, the current config files are not suitable for novices. We can easily get the mean of any parameter from the comment in Pytorch by jumping functions, but it is hard for the configs without comments.

**Reply by liguopeng0923** [reply]
Comments for common parameters are necessary for me. This is important for knowing how they work.

**Reply by yoshitomo-matsubara** [reply, marked as answer]
Hi @liguopeng0923 

I still don't understand what you want to do using torchdistill. Without the context, I cannot give you a concrete answer.

Note that you're seeing configs for torchdistill v1.0.0-dev as you refer to `key` and `init`. The torchdistill v1.0.0 is not available yet but will be released soon. If you want to test something right now, please read https://github.com/yoshitomo-matsubara/torchdistill#important-notice . This is very important.

For common parameters, you can see `datasets`, `models`, `train`, `test`as "common" parameters used in example scripts I shared in this repo.
Imagine your executable script `examples/legacy/image_classification.py` as a pipeline with a lot of abstracted modules, and your PyYAML config file e.g., https://github.com/yoshitomo-matsubara/torchdistill/blob/main/configs/sample/cifar10/ce/resnet20-final_run.yaml will help the script replace the abstracted modules with instantiated modules / functions.

I highly recommend you start with some examples e.g., https://github.com/yoshitomo-matsubara/torchdistill#cifar-10-and-cifar-100 using torchdistill v0.3.3.
If you are not familiar with PyTorch or PyYAML, you need to understand how they work and what types of modules are commonly used. 
For instance, if you see [torchvision's script](https://github.com/pytorch/vision/tree/main/references/classification) but cannot understand any components/designs, it may be difficult for you to see the design of the example script

**Reply by yoshitomo-matsubara** [reply]
P.S. Use "Reply" to the original reply for discussion to keep conversations in one thread

**Comment by liguopeng0923** [comment]
I get it. Thank you ! ! !
